src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -4573,6 +4549,35 @@ class CreateTableStatementSegment(ansi.CreateTableStatementSegment):
             OneOf(
                 # Create AS syntax:
                 Sequence(
+                    AnySetOf(
```

Comment:
dynamic table properties does not follow any strict order and can be in any sequence 

also these properties has to be before `as <query>`